src/Raven.Server/Utils/ShardHelper.cs

```diff
@@ -323,5 +338,19 @@ private static void ValidateBucketsMapping(DatabaseRecord record)
                 lastShard = current.ShardNumber;
             }
         }
+
+        public static string GenerateStickyId(string id, int bucket, char identityPartsSeparator)
+        {
+            Debug.Assert(id[^1] == identityPartsSeparator, "id[^1] == identityPartsSeparator");
+
+            return $"{id}${bucket}{SuffixIdTerminator}";
+        }
+
+        public static unsafe void ExtractStickyId(ref char* buffer, ref int size)
+        {
+            AdjustAfterSeparator('$', ref buffer, ref size);
+        }
+
+        public const char SuffixIdTerminator = '*';
```

Comment:
That is a currently a valid document id char, no?
What about doing something like `users/$12312/` And then we can detect the `/$` previously and react